test/hotspot/jtreg/runtime/cds/appcds/DumpClassListWithLF.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @summary Tests the format checking of LF_RESOLVE in classlist.
+ *
+ * @requires vm.cds
+ * @library /test/lib /test/hotspot/jtreg/runtime/cds/appcds
+ * @compile ClassListFormatBase.java test-classes/Hello.java
+ * @run driver DumpClassListWithLF
+ */
+
+public class DumpClassListWithLF extends ClassListFormatBase {
+    static final String MESSAGE_OK = "Replaced class java/lang/invoke/DirectMethodHandle$Holder";
+    static final String MESSAGE_NOT_OK = "Failed call to java/lang/invoke/GenerateJLIClassesHelper.cdsGenerateHolderClasses";
+
+    public static void main(String[] args) throws Throwable {
+
+        String appJar = JarBuilder.getOrCreateHelloJar();
+        //
+        // Note the class regeneration via java/lang/invoke/GenerateJLIClassesHelper.cdsGenerateHolderClasses(String[] lines)
+        // Whether the regeneration successes or fails, the dump should pass. Only the message can be checked for result.
+        //
+        // 1. The two lines are copied from build default_jli.txt which will be add to class generating list as additional (besides the default)
+        //  classlist), but will be ignored since they will be added to the (methodType) set which already contain them.
+        dumpShouldPass(
+            "TESTCASE 1: With correct output format",
+            appJar, classlist(
+                "Hello",
+                "@lambda-form-invoker [LF_RESOLVE] java.lang.invoke.DirectMethodHandle$Holder invokeStatic L7_L (success)",
+                "@lambda-form-invoker [LF_RESOLVE] java.lang.invoke.DirectMethodHandle$Holder invokeStatic LL_I (success)"),
```

Comment:
I think the `(success)` part should not be included in the classlist. The classlist is the public interface. It should be concise and include only the necessary information.

```
@lambda-form-invoker [LF_RESOLVE] java.lang.invoke.DirectMethodHandle$Holder invokeStatic L7_L
@lambda-form-invoker [LF_RESOLVE] java.lang.invoke.DirectMethodHandle$Holder invokeStatic LL_I 
```